Pod/Classes/CCNMigrationQueue.m

```diff
@@ -44,27 +45,65 @@ - (instancetype)init
 	return nil;
 }
 
+- (BOOL)checkForExecutions:(CIRDatabase *)database
+{
+	[self calculateExecutionsCountForDatabase:database];
+
+	return _executionsCount > 0;
+}
+
+- (NSUInteger)executionsCountForDatabase:(CIRDatabase *)database
+{
+	[self calculateExecutionsCountForDatabase:database];
+
+	return _executionsCount;
+}
+
+- (void)calculateExecutionsCountForDatabase:(CIRDatabase *)database
+{
+	NSUInteger count = 0;
```

Comment:
I think it's better to initialize the `_executionsCount` with -1, and them only execute the count query if the value of field is == -1. Or... this class holds a weak ref for the database, when it is set, run the count query and this two methods only return the field. I think the first solution is simple, therefore, better.